lib/ansible/modules/lineinfile.py

```diff
@@ -564,13 +623,13 @@ def main():
         if ins_bef is None and ins_aft is None:
             ins_aft = 'EOF'
 
-        present(module, path, regexp, line,
+        present(module, path, regexp, literal, line,
                 ins_aft, ins_bef, create, backup, backrefs, firstmatch)
     else:
-        if regexp is None and line is None:
-            module.fail_json(msg='one of line or regexp is required with state=absent')
+        if regexp is None and literal is None and line is None:
```

Comment:
Same suggestion here:
```suggestion
        if (regexp, literal, line) == (None, None, None):
```